machines/vic/reification/appliance.go

```diff
@@ -0,0 +1,416 @@
+				return vm.PowerOff(ctx)
+			})
+			if err != nil {
+				err = errors.Errorf("Failed to power off existing appliance: %s", err)
+				return err
+			}
+			_, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+				return vm.Destroy(ctx)
+			})
+			if err != nil {
+				err = errors.Errorf("Failed to destroy existing appliance: %s", err)
+				return err
+			}
+			m := object.NewFileManager(d.session.Client.Client)
+			if _, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+				return m.DeleteDatastoreFile(ctx, d.session.Datastore.Path(conf.DisplayName), d.session.Datacenter)
+			}); err != nil {
+				err = errors.Errorf("Failed to remove existing VCH data files, %s", err)
+				return err
+			}
+		} else if vm != nil {
+			err = errors.Errorf("VM already exists with display name %s. Name must be unique. Exiting...", conf.DisplayName)
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *Dispatcher) getNetworkDevices(conf *config.Config) ([]types.BaseVirtualDeviceConfigSpec, error) {
+	var devices []types.BaseVirtualDeviceConfigSpec
+	var err error
+
+	// bridge network
+	network, err := d.session.Finder.NetworkOrDefault(d.ctx, conf.BridgeNetworkPath)
+	if err != nil {
+		err = errors.Errorf("Failed to get bridge network: %s", err)
+		return nil, err
+	}
+	d.networks["bridge"] = network
+	d.nics[conf.BridgeNetworkName] = "bridge"
+
+	// client network
+	network, err = d.session.Finder.NetworkOrDefault(d.ctx, conf.ExtenalNetworkPath)
+	if err != nil {
+		err = errors.Errorf("Failed to get external network: %s", err)
+		return nil, err
+	}
+	d.networks["client"] = network
+	d.nics[conf.ExtenalNetworkName] = "client"
+
+	// management network
+	if conf.ManagementNetworkName != "" {
+		network, err = d.session.Finder.NetworkOrDefault(d.ctx, conf.ManagementNetworkPath)
+		if err != nil {
+			err = errors.Errorf("Failed to get management network: %s", err)
+			return nil, err
+		}
+		d.networks["management"] = network
+		d.nics[conf.ManagementNetworkName] = "management"
+	}
+
+	var backing types.BaseVirtualDeviceBackingInfo
+	for _, network := range d.networks {
+		backing, err = network.EthernetCardBackingInfo(d.ctx)
+		if err != nil {
+			err = errors.Errorf("Failed to get network backing info for %s: %s", network, err)
+			return nil, err
+		}
+
+		net := &types.VirtualDeviceConfigSpec{
+			Operation: types.VirtualDeviceConfigSpecOperationAdd,
+			Device: &types.VirtualVmxnet3{
+				VirtualVmxnet: types.VirtualVmxnet{
+					VirtualEthernetCard: types.VirtualEthernetCard{
+						VirtualDevice: types.VirtualDevice{
+							Key:     -3,
+							Backing: backing,
+						},
+						AddressType: string(types.VirtualEthernetCardMacTypeGenerated),
+					},
+				},
+			},
+		}
+		devices = append(devices, net)
+	}
+	return devices, nil
+}
+
+func (d *Dispatcher) createApplianceSpec(conf *config.Config) (*types.VirtualMachineConfigSpec, error) {
```

Comment:
I thought about that, but as George said, the spec package will need to be refactored as well.